default/python/AI/colonization/calculate_production.py

```diff
@@ -47,47 +54,45 @@ def _get_modified_industry_bonuses():
     centre_bonus1 = get_named_real("BLD_INDUSTRY_CENTER_1_TARGET_INDUSTRY_PERPOP")
     centre_bonus2 = get_named_real("BLD_INDUSTRY_CENTER_2_TARGET_INDUSTRY_PERPOP")
     centre_bonus3 = get_named_real("BLD_INDUSTRY_CENTER_3_TARGET_INDUSTRY_PERPOP")
-    # use fo.getEmpire().researchQueue directly here to simplify caching this function
-    top3_research = [fo.getEmpire().researchQueue[i] for i in range(0, 3)]
     return [
         _ProductionBonus(
-            tech_is_complete("PRO_FUSION_GEN") or "PRO_FUSION_GEN" in top3_research,
+            tech_soon_available("PRO_FUSION_GEN", 3),
```

Comment:
This is a nice change. 

Also, this is a good example of why having small focused PR is great. 

Having it in a separate PR with a comment, that this is refactoring and small would make it merge faster.  
I always could spend a minute on small reviews when I browse my mail.

In general, it's ok to have some cleanup together with features, but I want to build the process when it happens very rarely. 

Backporting changes from the development branch with git and Pycharm is not time-consuming. Maybe we should meet and I'll screen share some tricks and tips related to that process.